leader-election-impl/src/test/java/com/palantir/leader/proxy/AwaitingLeadershipProxyTest.java

```diff
@@ -333,6 +339,148 @@ public void shouldClearDelegateIfLeadershipTokenIsRefreshedByAnotherProxy() thro
+                    throw new SuspectedNotCurrentLeaderException("There is one imposter among us");
+                })
+                .when(mock)
+                .val();
+
+        assertThatLoggableExceptionThrownBy(proxy::val)
+                .isInstanceOf(NotCurrentLeaderException.class)
+                .hasMessageContaining("method invoked on a non-leader (leadership lost)");
+
+        // Another call is required before we clear existing resources.
+        assertThatLoggableExceptionThrownBy(proxy::val)
+                .isInstanceOf(NotCurrentLeaderException.class)
+                .hasMessageContaining("method invoked on a non-leader");
+        verify(mock).close();
+    }
+
+    @Test
+    public void shouldLoseLeadershipOnSuspectedNotCurrentLeaderWhenNoQuorum() throws InterruptedException, IOException {
+        MyCloseable mock = mock(MyCloseable.class);
+        MyCloseable proxy = AwaitingLeadershipProxy.newProxyInstance(
+                MyCloseable.class, () -> mock, LeadershipCoordinator.create(leaderElectionService));
+        waitForLeadershipToBeGained();
+
+        doAnswer(_invocation -> {
+                    loseLeadershipOnLeaderElectionService(StillLeadingStatus.NO_QUORUM);
+                    throw new SuspectedNotCurrentLeaderException("There is one imposter among us");
+                })
+                .when(mock)
+                .val();
+
+        assertThatLoggableExceptionThrownBy(proxy::val)
+                .isInstanceOf(NotCurrentLeaderException.class)
+                .hasMessageContaining("method invoked on a non-leader (leadership lost)");
+
+        // Another call is required before we clear existing resources.
+        assertThatLoggableExceptionThrownBy(proxy::val)
+                .isInstanceOf(NotCurrentLeaderException.class)
+                .hasMessageContaining("method invoked on a non-leader");
+        verify(mock).close();
+    }
+
+    @Test
+    public void shouldLoseLeadershipOnSuspectedNotCurrentLeaderWhenFailingToCheckLeadershipState()
+            throws InterruptedException, IOException {
+        MyCloseable mock = mock(MyCloseable.class);
+        MyCloseable proxy = AwaitingLeadershipProxy.newProxyInstance(
+                MyCloseable.class, () -> mock, LeadershipCoordinator.create(leaderElectionService));
+        waitForLeadershipToBeGained();
+
+        doAnswer(_invocation -> {
+                    throw new SuspectedNotCurrentLeaderException("There is one imposter among us");
+                })
+                .when(mock)
+                .val();
+
+        RuntimeException underlyingError = new RuntimeException("Something went wrong");
+        when(leaderElectionService.isStillLeading(any()))
+                .thenReturn(Futures.immediateFuture(StillLeadingStatus.LEADING))
+                .thenReturn(Futures.immediateFailedFuture(underlyingError))
+                .thenReturn(Futures.immediateFuture(StillLeadingStatus.NOT_LEADING));
```

Comment:
optional nit: consider leaving a comment as to why we actually need to return a status eventually for this test to succeed (it was not immediately obvious, since this test was about failing to check leadership state!)

Reply:
yeah I'll fix it